src/process/process_set.cu

```diff
@@ -4,71 +4,92 @@
-        size_t* product_ids_,
-        double* yields_)
+        size_t n_species)
+       
     {
       // define thread index
       size_t tid = blockIdx.x * blockDim.x + threadIdx.x;
       size_t react_id_offset, prod_id_offset, yield_offset;
-
+      double* forcing = device->forcing; 
+      size_t* number_of_reactants = device->number_of_reactants; 
+      size_t* reactant_ids = device->reactant_ids; 
+      size_t* number_of_products = device->number_of_products; 
       if (tid < n_grids)
       {
         react_id_offset = 0;
         prod_id_offset = 0;
         yield_offset = 0;
         for (std::size_t i_rxn = 0; i_rxn < n_reactions; ++i_rxn)
         {
-          double rate = rate_constants[i_rxn * n_grids + tid];
-          for (std::size_t i_react = 0; i_react < number_of_reactants_[i_rxn]; ++i_react)
-            rate *= state_variables[reactant_ids_[react_id_offset + i_react] * n_grids + tid];
-          for (std::size_t i_react = 0; i_react < number_of_reactants_[i_rxn]; ++i_react)
+          double rate = device->rate_constants[i_rxn * n_grids + tid];
+          for (std::size_t i_react = 0; i_react < number_of_reactants[i_rxn]; ++i_react)
+            rate *= device->state_variables[reactant_ids[react_id_offset + i_react] * n_grids + tid];
+          for (std::size_t i_react = 0; i_react < number_of_reactants[i_rxn]; ++i_react)
           {
-            forcing[reactant_ids_[react_id_offset + i_react] * n_grids + tid] -= rate;
+            forcing[reactant_ids[react_id_offset + i_react] * n_grids + tid] -= rate;
           }
-          for (std::size_t i_prod = 0; i_prod < number_of_products_[i_rxn]; ++i_prod)
+          for (std::size_t i_prod = 0; i_prod < number_of_products[i_rxn]; ++i_prod)
           {
-            size_t index = product_ids_[prod_id_offset + i_prod] * n_grids + tid;
-            forcing[index] += yields_[yield_offset + i_prod] * rate;
+            size_t index = device->product_ids[prod_id_offset + i_prod] * n_grids + tid;
+            forcing[index] += device->yields[yield_offset + i_prod] * rate;
           }
-          react_id_offset += number_of_reactants_[i_rxn];
-          prod_id_offset += number_of_products_[i_rxn];
-          yield_offset += number_of_products_[i_rxn];
+          react_id_offset += number_of_reactants[i_rxn];
+          prod_id_offset += number_of_products[i_rxn];
+          yield_offset += number_of_products[i_rxn];
         }  // for loop over number of reactions
       }    // if check for valid CUDA threads
     }      // end of AddForcingTerms_kernel
 
     __global__ void AddJacobianTermsKernel(
-        double* rate_constants,
-        double* state_variables,
+        jacobianDevice* device,
         size_t n_grids,
-        size_t n_reactions,
-        double* jacobian,
-        size_t* number_of_reactants,
-        size_t* reactant_ids,
-        size_t* number_of_products,
-        double* yields,
-        size_t* jacobian_flat_ids)
+        size_t n_reactions)
```

Comment:
would it make sense to also include these in the `JacobianDevice` struct?

Reply:
changes made.